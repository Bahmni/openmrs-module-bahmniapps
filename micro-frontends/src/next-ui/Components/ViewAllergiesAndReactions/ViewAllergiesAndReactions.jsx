import React from "react";
import "./ViewAllergiesAndReactions.scss";
import "../../../styles/common.scss";
import {FormattedMessage} from "react-intl";
import { Accordion, AccordionItem } from "carbon-components-react";
import { Document } from "@carbon/icons-react/next";
import PropTypes from "prop-types";
export const ViewAllergiesAndReactions = (props) => {
    const { allergies, showTextAsAbnormal } = props;
    return <div className={"next-ui"}>
            <div className={"allergies-row allergies-row-heading"}>
                <div><FormattedMessage id={"ALLERGEN"} defaultMessage={"Allergen"}/></div>
                <div><FormattedMessage id={"REACTIONS"} defaultMessage={"Reaction(s)"}/></div>
                <div><FormattedMessage id={"SEVERITY"} defaultMessage={"Severity"}/></div>
            </div>
            <div>
                {allergies.map((allergy, index) => {
                    const title = <div key={index}
                                       className={` allergies-row ${showTextAsAbnormal ? "allergies-red-text" 
                                           : allergy.severity === "severe" ? "allergies-red-text": ""}`}>
                        <div>{allergy.allergen}</div>
                        <div>{allergy.reactions.join(", ")}</div>
                        <div className={"capitalize"}>{allergy.severity}</div>
                    </div>
                    return (<Accordion>
                        <AccordionItem title={title}>
                            <div className={"allergies-accordion-item"}>
                                <div>
                                    {allergy.note &&
                                        <div className={"allergies-row-note"}><Document size={20}/>
                                            {allergy.note}</div>}
                                </div>
                                <div className={"allergy-provider"}>
                                    {allergy.provider}
                                </div>
                            </div>
                        </AccordionItem>
                    </Accordion>)
                })}
            </div>
        </div>
}

ViewAllergiesAndReactions.propTypes = {
    allergies: PropTypes.array.isRequired,
    showTextAsAbnormal: PropTypes.bool,
}