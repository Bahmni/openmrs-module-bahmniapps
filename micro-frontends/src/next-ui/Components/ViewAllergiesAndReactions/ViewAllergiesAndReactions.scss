.allergies-row{
  display: grid;
  grid-template-columns: 1fr 2fr 1fr;
  gap: 10px;
}
.allergies-row-heading{
  font-weight: 600;
  background-color: #F1F1F1;
  padding: 0 8px;
  grid-template-columns: 1fr 2fr 1fr 12px;
  div{
    padding: 10px 0;
  }
}
.allergies-row-note{
  display: flex;
  align-items: center;
  padding: 10px;
  gap: 5px;
  svg{
    min-width: 20px;
  }
}
.allergies-red-text{
  color: red
}
.bx--accordion__content{
  padding: 0;
}
.allergies-accordion-item{
  display: flex;
  justify-content: space-between;
  background-color: #EFF8FF;
  align-items: center;
}
.allergy-provider{
  padding: 10px;
}
