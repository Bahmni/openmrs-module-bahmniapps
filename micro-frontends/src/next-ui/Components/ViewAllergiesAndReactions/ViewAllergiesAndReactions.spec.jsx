import React from "react";
import { render } from "@testing-library/react";
import { ViewAllergiesAndReactions } from './ViewAllergiesAndReactions';

describe("ViewAllergiesAndReactions", () => {
    const allergiesMock = [
        {
            allergen: "Bee",
            reactions: ["Hives", "Itching", "Fever"],
            severity: "severe",
            provider: "Dr. John Doe"
        },
        {
            allergen: "Peanuts",
            reactions: ["Dizziness", "Fever"],
            severity: "mild",
            note: "Onset Date: 2023-10-01",
            provider: "Dr. Jane"
        }
    ]
    it('should render ViewAllergiesAndReactions component', () => {
        const {container} = render(<ViewAllergiesAndReactions allergies={allergiesMock}/>);
        expect(container).toMatchSnapshot();
    });
});