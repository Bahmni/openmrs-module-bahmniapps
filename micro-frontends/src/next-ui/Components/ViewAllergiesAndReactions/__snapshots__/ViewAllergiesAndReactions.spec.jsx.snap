// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewAllergiesAndReactions should render ViewAllergiesAndReactions component 1`] = `
<div>
  <div
    class="next-ui"
  >
    <div
      class="allergies-row allergies-row-heading"
    >
      <div>
        Allergen
      </div>
      <div>
        Reaction(s)
      </div>
      <div>
        Severity
      </div>
    </div>
    <div>
      <ul
        class="bx--accordion bx--accordion--end bx--accordion--md"
      >
        <li
          class="bx--accordion__item"
        >
          <button
            aria-controls="accordion-item-1"
            aria-expanded="false"
            class="bx--accordion__heading"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="bx--accordion__arrow"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
              />
            </svg>
            <div
              class="bx--accordion__title"
              dir="auto"
            >
              <div
                class=" allergies-row allergies-red-text"
              >
                <div>
                  Bee
                </div>
                <div>
                  Hives, Itching, Fever
                </div>
                <div
                  class="capitalize"
                >
                  severe
                </div>
              </div>
            </div>
          </button>
          <div
            class="bx--accordion__content"
            id="accordion-item-1"
          >
            <div
              class="allergies-accordion-item"
            >
              <div />
              <div
                class="allergy-provider"
              >
                Dr. John Doe
              </div>
            </div>
          </div>
        </li>
      </ul>
      <ul
        class="bx--accordion bx--accordion--end bx--accordion--md"
      >
        <li
          class="bx--accordion__item"
        >
          <button
            aria-controls="accordion-item-2"
            aria-expanded="false"
            class="bx--accordion__heading"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="bx--accordion__arrow"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
              />
            </svg>
            <div
              class="bx--accordion__title"
              dir="auto"
            >
              <div
                class=" allergies-row "
              >
                <div>
                  Peanuts
                </div>
                <div>
                  Dizziness, Fever
                </div>
                <div
                  class="capitalize"
                >
                  mild
                </div>
              </div>
            </div>
          </button>
          <div
            class="bx--accordion__content"
            id="accordion-item-2"
          >
            <div
              class="allergies-accordion-item"
            >
              <div>
                <div
                  class="allergies-row-note"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    height="20"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M25.7,9.3l-7-7C18.5,2.1,18.3,2,18,2H8C6.9,2,6,2.9,6,4v24c0,1.1,0.9,2,2,2h16c1.1,0,2-0.9,2-2V10C26,9.7,25.9,9.5,25.7,9.3 z M18,4.4l5.6,5.6H18V4.4z M24,28H8V4h8v6c0,1.1,0.9,2,2,2h6V28z"
                    />
                    <path
                      d="M10 22H22V24H10zM10 16H22V18H10z"
                    />
                  </svg>
                  Onset Date: 2023-10-01
                </div>
              </div>
              <div
                class="allergy-provider"
              >
                Dr. Jane
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
